lib/widgets/message_list.dart

```diff
@@ -849,61 +852,74 @@ class MessageWithPossibleSender extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final message = item.message;
-    final time = _kMessageTimestampFormat
-      .format(DateTime.fromMillisecondsSinceEpoch(1000 * message.timestamp));
 
-    return GestureDetector(
-      behavior: HitTestBehavior.translucent,
-      onLongPress: () => showMessageActionSheet(context: context, message: message),
-      // TODO clean up this layout, by less precisely imitating web
-      child: Padding(
-        padding: const EdgeInsets.only(top: 2, bottom: 3, left: 8, right: 8),
-        child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
-          item.showSender
-            ? Padding(
-                padding: const EdgeInsets.fromLTRB(3, 6, 11, 0),
+    final Widget? senderRow = item.showSender
+      ? Padding(
+          padding: const EdgeInsets.only(top: 2, bottom: 4, left: 16, right: 16),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
```

Comment:
This `spaceBetween` is missing in the commit
1d3d41d20 msglist: Consolidate sender avatar, name and message time into same row

and as a result the timestamp goes immediately after the sender name:
![image](https://github.com/zulip/zulip-flutter/assets/28173/4104f57c-02da-4e3f-9c22-1f3e542166d8)

Reply:
More generally, it seems like a lot of what this later commit:
53edaad70 msglist: Update message sender style and layout

is doing is just fixing bugs in the earlier commit — things that became necessities as a consequence of creating this `Row` for the sender avatar, name, and timestamp.

The lack of `spaceBetween` is the most visible, but for another example: similarly to `spaceBetween`, if we're going to have the name and the timestamp in one Row then it seems like we naturally want `CrossAxisAlignment.baseline` for that. And we need `Flexible` and `TextOverflow.ellipsis` so that things don't break if a sender has a long name.

As discussed in Zulip's general commit style guide:
https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#each-commit-must-be-coherent
each commit should leave things in a state it would be reasonable for us to deploy as is. (I.e. to release as is, to use the verb that's applicable for a client.) So if part of a PR introduces a bug and then a later part fixes the bug, the fix should instead get squashed into the commit that introduced it.